python/GafferImageTest/MixTest.py

```diff
@@ -0,0 +1,346 @@
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import os
+import unittest
+
+import IECore
+
+import Gaffer
+import GafferTest
+import GafferImage
+import GafferImageTest
+
+class MixTest( GafferImageTest.ImageTestCase ) :
+
+	rPath = os.path.expandvars( "$GAFFER_ROOT/python/GafferImageTest/images/redWithDataWindow.100x100.exr" )
+	gPath = os.path.expandvars( "$GAFFER_ROOT/python/GafferImageTest/images/greenWithDataWindow.100x100.exr" )
+	checkerPath = os.path.expandvars( "$GAFFER_ROOT/python/GafferImageTest/images/checkerboard.100x100.exr" )
+	checkerMixPath = os.path.expandvars( "$GAFFER_ROOT/python/GafferImageTest/images/checkerMix.100x100.exr" )
+
+	# Do several tests to check the cache is working correctly:
+	def testHashes( self ) :
+
+		r1 = GafferImage.ImageReader()
+		r1["fileName"].setValue( self.checkerPath )
+
+		r2 = GafferImage.ImageReader()
+		r2["fileName"].setValue( self.gPath )
+
+		##########################################
+		# Test to see if the hash changes.
+		##########################################
+
+		mix = GafferImage.Mix()
+
+		mix["in"][0].setInput( r1["out"] )
+		mix["in"][1].setInput( r2["out"] )
+		h1 = mix["out"].image().hash()
+
+		# Switch the inputs.
+		mix["in"][1].setInput( r1["out"] )
+		mix["in"][0].setInput( r2["out"] )
+		h2 = mix["out"].image().hash()
+
+		self.assertNotEqual( h1, h2 )
+
+	def testHashPassThrough( self ) :
+
+		r1 = GafferImage.ImageReader()
+		r1["fileName"].setValue( self.checkerPath )
+
+		##########################################
+		# Test to see if the input has is always passed
+		# through if only the first input is connected.
+		##########################################
+
+		mix = GafferImage.Mix()
+
+		expectedHash = r1["out"].image().hash()
```

Comment:
This isn't testing what it claims to be testing. We want to be comparing the Gaffer hashes that will be used to share entries in Gaffer's cache (`r1["out"].imageHash()`), but instead we're computing the image and comparing the hash of the data we get. Since this is copy pasted from MergeTest, any chance you could fix that too?